lib/uplink/uplink.go

```diff
@@ -4,10 +4,84 @@
 package uplink
 
 import (
-	"github.com/zeebo/errs"
+	"context"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/eestream"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/metainfo/kvmetainfo"
+	"storj.io/storj/pkg/peertls/tlsopts"
+	"storj.io/storj/pkg/storage/buckets"
+	"storj.io/storj/pkg/storage/segments"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/uplink/metainfo"
 )
 
 var (
-	// Error is the errs class of standard End User Client errors
-	Error = errs.Class("libuplink error")
+	maxBucketMetaSize = 10 * memory.MiB
 )
+
+// Config represents configuration options for an Uplink
+type Config struct {
+	// These config values are likely to change semantics or go away
+	// entirely between releases. Be careful when using them!
+	Volatile struct {
+		TLS struct {
+			SkipPeerCAWhitelist bool
+		}
+	}
+}
+
+// Uplink represents the main entrypoint to Storj V3. An Uplink connects to
+// a specific Satellite and caches connections and resources, allowing one to
+// create sessions delineated by specific access controls.
+type Uplink struct {
+	tc transport.Client
+}
+
+// NewUplink creates a new Uplink
+func NewUplink(ctx context.Context, cfg Config) (*Uplink, error) {
```

Comment:
How do we plan to handle configs here?

Reply:
good question. maybe the library should be able to read a config file, if asked, like something called `uplink.ReadConfigFromFile()`?

Reply:
That would work with the cmd util. Do we still want per-bucket configs as a possibility?

Reply:
I don't know. This is difficult for me to judge without seeing several "real life" uses of the library. I want to avoid putting something in here prematurely that doesn't meet user needs in quite the right way, and then end up being required to maintain it for a long time.

It would probably be better to put config handling in the cmd/uplink code for now, and then if we see a need, move it into the library.